src/lightning_app/runners/cloud.py

```diff
@@ -111,6 +112,11 @@ def dispatch(
             # Override the cluster ID if provided by the CLI
             app_config.cluster_id = cluster_id
 
+        if ENABLE_APP_CHECKPOINT and self.checkpoint:
+            app_config.checkpoint = self.checkpoint
+        else:
+            app_config.checkpoint = None
```

Comment:
```suggestion
        if ENABLE_APP_CHECKPOINT and self.checkpoint:
            app_config.checkpoint = self.checkpoint
```
You already added the default to the config